server/schema.js

```diff
@@ -1,36 +1,79 @@
 const { gql } = require("apollo-server");
+const _ = require("lodash");
+const { makeExecutableSchema } = require("@graphql-tools/schema");
+const { mapSchema, getDirective, MapperKind } = require("@graphql-tools/utils");
+const { defaultFieldResolver } = require("graphql");
+const resolvers = require("./resolvers");
 
-const typeDefs = gql`
-"Image url for a breed"
-type BreedImage{
-  id: ID!
-  url: String!
+// This function takes in a schema and adds snake-casing logic
+// to every resolver for an object field that has a directive with
+// the specified name (we're using `snakeCase`)
+function snakeCaseDirectiveTransformer(schema, directiveName) {
+  return mapSchema(schema, {
+    // Executes once for each object field in the schema
+    [MapperKind.OBJECT_FIELD]: (fieldConfig) => {
+      // Check whether this field has the specified directive
+      const snakeCaseDirective = getDirective(
+        schema,
+        fieldConfig,
+        directiveName
+      )?.[0];
+
+      if (snakeCaseDirective) {
+        // Get this field's original resolver
+        const { resolve = defaultFieldResolver } = fieldConfig;
+
+        fieldConfig.resolve = (source, args, context, info) => {
+          return source[_.snakeCase(info.fieldName)]
+        }
+
+        return fieldConfig;
+      }
+    },
+  });
 }
```

Comment:
That seems too smart. 🤔 

Also, it's a bit weird from a GraphQL perspective. There is a `snake_case` directive defined in the GQL schema which actually is used to map data taken from a 3rd party API and match it to the schema itself.

If the 3rd party is changed (or starts provided a `camelCase`?), would you change the GQL schema?